### ISSUE TITLE
Fix S3928xxxx FP: Argument validation in lambda

### ISSUE THREAD

**Issue by paulhickman-ec** [SonarSource/sonar-dotnet#8319]
### Description

S3928 fires if the ArgumentException is thrown from within a lambda function.

### Repro steps

````c#
    public class Foo
    {
        public void Bar(int x)
        {
            Wrapper(() =>
            {
                throw new ArgumentException("blah", nameof(x));
            });
        }

        static void Wrapper(Action action) => action();
    }
````

### Expected behavior

No S3928 warning is produced

### Actual behavior

S3928: The parameter name 'x' is not declared in the argument list

### Known workarounds

Disable the warning with a #pragma

### Related information

* SonarLInk for Visual Studio 2022: 7.3.0.77872
* Visual Studio 2022: 17.7.6
* SonarScanner: 9.12.0.78982
* Operating System: Windows 11

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @paulhickman-ec, 

The current implementation of the rule is designed to examine the immediate parent definition. In this instance, it's the `ParenthesizedLambdaExpressionSyntax`, which doesn't have any parameters, hence the issue is raised.
I found similar test cases in our repo:
```cs
void Foo7(int a)
{
    Action<int> res =
        i =>
        {
            throw new ArgumentNullException("i");
            throw new ArgumentNullException("a"); // Noncompliant 
        };
}
[...]
public void ValidationDoneByLocalMethod(string methodArg)
{
    ValidateLocal();

    void ValidateLocal()
    {
        if (methodArg == null)
        {
            throw new ArgumentNullException(nameof(methodArg)); // Noncompliant
        }
    }
}
```
Upon reviewing these examples and the one you provided, I agree that they should be considered as FP. I will proceed to mark them as such and will also include your example for further reference.
Thank you

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1652](https://sonarsource.atlassian.net/browse/NET-1652)

[NET-1652]: https://sonarsource.atlassian.net/browse/NET-1652?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
